MantidQt/MantidWidgets/inc/MantidQtMantidWidgets/WorkspacePresenter/WorkspacePresenter.h

```diff
@@ -0,0 +1,73 @@
+#ifndef MANTID_MANTIDWIDGETS_WORKSPACEPRESENTER_H_
+#define MANTID_MANTIDWIDGETS_WORKSPACEPRESENTER_H_
+
+#include "MantidQtMantidWidgets/WorkspacePresenter/IWorkspacePresenter.h"
+#include <memory>
+#include <boost/weak_ptr.hpp>
+
+namespace MantidQt {
+namespace MantidWidgets {
+
+class IWorkspaceDockView;
+class IADSAdapter;
+
+using DockView_wptr = boost::weak_ptr<IWorkspaceDockView>;
+using ADSAdapter_uptr = std::unique_ptr<IADSAdapter>;
+/**
+\class  WorkspacePresenter
+\brief  Presenter class for Workspace dock in MantidPlot UI
+\author Lamar Moore
+\date   24-08-2016
+\version 1.0
+
+
+Copyright &copy; 2016 ISIS Rutherford Appleton Laboratory, NScD Oak Ridge
+National Laboratory & European Spallation Source
+
+This file is part of Mantid.
+
+Mantid is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 3 of the License, or
+(at your option) any later version.
+
+Mantid is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+File change history is stored at: <https://github.com/mantidproject/mantid>
+*/
+class WorkspacePresenter : public IWorkspacePresenter {
+public:
+  explicit WorkspacePresenter(DockView_wptr view, ADSAdapter_uptr adapter);
+  ~WorkspacePresenter() = default;
+
+  void notify(ADSNotifiable::Flag flag);
```

Comment:
Yes, this is exactly what I thought you'd do. I might suggest that the individual notify methods should be named in such a way that allows them to be distinguished. i.e. `notifyFromView` and `notifyFromWorkspaceSource` or something similar.